lib/fluent/plugin/filter_elasticsearch_timestamp_check.rb

```diff
@@ -32,13 +36,13 @@ def filter(tag, time, record)
             record['@timestamp'] = record['fluent_converted_timestamp'] =
               Time.at(
                 num / (10 ** ((Math.log10(num).to_i + 1) - 10))
-              ).strftime('%Y-%m-%dT%H:%M:%S.%L%z')
+              ).strftime(@strftime_format)
             break
           end
 
           # normal timestamp string processing
           record['@timestamp'] = record['fluent_converted_timestamp'] =
-            DateTime.parse(timestamp).strftime('%Y-%m-%dT%H:%M:%S.%L%z')
+            DateTime.parse(timestamp).strftime(@strftime_format)
```

Comment:
has the behavior of elasticsearch been validated for encountering sub-millisecond precision timestamp in the @timestamp field? i.e. the behavior we want (and hopefully elasticsearch does) is that it simply truncate the precision to millisecond, but it could also behave in such way that it might throw away the millisecond field altogether once higher precision is encountered (bad but not catastrophic) or index them badly which causes the entry not retrievable in the future.

Reply:
elasticsearch will simply truncate micro or nano or higher precision time into millisecond for now.
But in [this elasticsearch issue](https://github.com/elastic/elasticsearch/issues/10005), developers and users discuss to support nanosecond precision time.
This PR aims to discuss for high resolution time support.

Reply:
👍